crates/sui-framework/packages/bridge/sources/committee.move

```diff
@@ -225,6 +330,191 @@ module bridge::committee {
+        assert_eq(5000, member1.voting_power);
+
+        test_utils::destroy(committee);
+        test_scenario::return_shared(system_state);
+        test_scenario::end(scenario);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = ESenderNotActiveValidator)]
+    fun test_init_committee_not_validator() {
+        let scenario = test_scenario::begin(@0x0);
+        let ctx = test_scenario::ctx(&mut scenario);
+        let committee = create(ctx);
+
+        let validators = vector[
+            create_validator_for_testing(@0xA, 100, ctx),
+            create_validator_for_testing(@0xC, 100, ctx)
+        ];
+        create_sui_system_state_for_testing(validators, 0, 0, ctx);
+        advance_epoch_with_reward_amounts(0, 0, &mut scenario);
+        test_scenario::next_tx(&mut scenario, @0x0);
+
+        let system_state = test_scenario::take_shared<SuiSystemState>(&scenario);
+
+        // validator registration
+        register(&mut committee, &mut system_state, hex::decode(VALIDATOR1_PUBKEY), b"", &tx(@0xD, 0));
+
+        test_utils::destroy(committee);
+        test_scenario::return_shared(system_state);
+        test_scenario::end(scenario);
+    }
+
+    #[test]
+    fun test_init_committee_validator_become_inactive() {
+        let scenario = test_scenario::begin(@0x0);
+        let ctx = test_scenario::ctx(&mut scenario);
+        let committee = create(ctx);
+
+        let validators = vector[
+            create_validator_for_testing(@0xA, 100, ctx),
+            create_validator_for_testing(@0xC, 100, ctx),
+            create_validator_for_testing(@0xD, 100, ctx),
+            create_validator_for_testing(@0xE, 100, ctx),
+            create_validator_for_testing(@0xF, 100, ctx)
+        ];
+        create_sui_system_state_for_testing(validators, 0, 0, ctx);
+        advance_epoch_with_reward_amounts(0, 0, &mut scenario);
+        test_scenario::next_tx(&mut scenario, @0x0);
+
+        let system_state = test_scenario::take_shared<SuiSystemState>(&scenario);
+
+        // validator registration, 3 validators registered, should have 60% voting power in total
+        register(&mut committee, &mut system_state, hex::decode(VALIDATOR1_PUBKEY), b"", &tx(@0xA, 0));
+        register(&mut committee, &mut system_state, hex::decode(VALIDATOR2_PUBKEY), b"", &tx(@0xC, 0));
+        register(&mut committee, &mut system_state, hex::decode(VALIDATOR3_PUBKEY), b"", &tx(@0xD, 0));
+
+        // Verify validator registration
+        assert_eq(3, vec_map::size(&committee.member_registrations));
+
+        // Validator 0xA become inactive, total voting power become 50%
+        sui_system::request_remove_validator(&mut system_state, &mut tx(@0xA, 0));
+        test_scenario::return_shared(system_state);
+        advance_epoch_with_reward_amounts(0, 0, &mut scenario);
+
+        let system_state = test_scenario::take_shared<SuiSystemState>(&scenario);
+
+        // create committee should not create a committe because of not enough stake.
```

Comment:
nit: create committee should fail because of not enough stake.

Reply:
This cannot fail because it’s part of the epoch end transaction, it will do nothing if not enough stake